CollapseLauncher/Classes/Helper/Database/DBHandler.cs

```diff
@@ -0,0 +1,304 @@
+                return c;
+            }
+            set
+            {
+                if (value != _userId) _isFirstInit = true; // Force first init if value changed
+                
+                _userId           = value;
+                DbConfig.UserGuid = value;
+                
+                var byteUidH = System.IO.Hashing.XxHash64.Hash(value.ToByteArray());
+                _userIdHash  = BitConverter.ToString(byteUidH).Replace("-", "").ToLowerInvariant();
+                _isFirstInit = true;
+            }
+        }
+
+        private static bool _isFirstInit = true;
+        #endregion
+
+        private static IDatabaseClient _database;
+
+        public static async Task Init(bool redirectThrow = false)
+        {
+            DbConfig.Init();
+            
+            if (!IsEnabled)
+            {
+                LogWriteLine("[DbHandler::Init] Database functionality is disabled!");
+                return;
+            }
+
+            try
+            {
+                // Init props
+                _ = Token;
+                _ = Uri;
+                _ = UserId;
+
+                if (string.IsNullOrEmpty(Uri))
+                    throw new NullReferenceException(Lang._SettingsPage.Database_Error_EmptyUri);
+                if (string.IsNullOrEmpty(Token))
+                    throw new NullReferenceException(Lang._SettingsPage.Database_Error_EmptyToken);
+
+                // Connect to database
+                // Libsql-client-dotnet technically support file based SQLite by pushing `file://` proto in the URL.
+                // But what's the point?
+                _database = await DatabaseClient.Create(opts =>
+                                                        {
+                                                            opts.Url       = Uri;
+                                                            opts.AuthToken = Token;
+                                                        });
+
+                if (_isFirstInit)
+                {
+                    LogWriteLine("[DbHandler::Init] Initializing database system...");
+                    // Ensure table exist at first initialization
+                    await
+                        _database
+                           .Execute($"CREATE TABLE IF NOT EXISTS \"uid-{_userIdHash}\" (Id INTEGER PRIMARY KEY AUTOINCREMENT, 'key' TEXT UNIQUE NOT NULL, 'value' TEXT)");
+                    _isFirstInit = false;
+                }
+                else LogWriteLine("[DbHandler::Init] Reinitializing database system...");
+            }
+            catch (Exception e) when (!redirectThrow)
+            {
+                LogWriteLine($"[DBHandler::Init] Error when (re)initializing database system!\r\n{e}", LogType.Error, true);
+            }
+            catch (Exception e)
+            {
+                LogWriteLine($"[DBHandler::Init] Error when (re)initializing database system!\r\n{e}", LogType.Error, true);
```

Comment:
This error will never be displayed as public because I want to set the database system to be completely in the background and non-invasive to the UX. I tried to do push in-app notifications but it seems currently the caller for it doesn't work outside HomePage methods (maybe i am dum, probably, most likely)